src/subtitle/subtitleTrack.ts

```diff
@@ -0,0 +1,60 @@
+import { MakeRequired } from '../utils';
+
+/**
+ * Supported subtitle/caption file formats.
+ */
+export enum SubtitleFormat {
+  CEA = 'cea',
+  TTML = 'ttml',
+  VTT = 'vtt',
+}
+
+/**
+ * Represents a custom subtitle track source that can be added to `SourceConfig.subtitleTracks`.
+ */
+export interface SubtitleTrack {
+  /**
+   * The URL to the timed file, e.g. WebVTT file.
+   */
+  url?: string;
+  /**
+   * The label for this track.
+   */
+  label?: string;
+  /**
+   * The unique identifier for this track. If no value is provided, a random UUIDv4 will be generated for it.
+   */
+  identifier?: string;
+  /**
+   * Specifies the file format to be used by this track.
+   */
+  format?: SubtitleFormat;
+  /**
+   * If set to true, this track would be considered as default. Default is `false`.
+   */
+  isDefault?: boolean;
+  /**
+   * Tells if a subtitle track is forced. If set to `true` it means that the player should automatically
+   * select and switch this subtitle according to the selected audio language. Forced subtitles do
+   * not appear in `Player.getAvailableSubtitles`.
+   *
+   * Default is `false`.
+   */
+  isForced?: boolean;
+  /**
+   * The IETF BCP 47 language tag associated with this track, e.g. `pt`, `en`, `es` etc.
+   */
+  language?: string;
+}
+
+/**
+ * Helper type that represents an entry in the `SourceConfig.addSubtitleTracks` list.
+ *
+ * Since `SubtitleTrack` has all of its properties as optionals for total compatibility with
+ * values that may be sent from native code, this type serves as a reinforcer of what properties
+ * should be required during the registration of an external subtitle track from JS.
+ */
+export type AddSubtitleTrack = MakeRequired<
+  SubtitleTrack,
+  'url' | 'label' | 'language'
+>;
```

Comment:
Do we absolutely need this?

Btw. on Android a `mimeType` is mandatory for side-loaded SubtitleTracks

--- 
If we need this, and it is only used in the `SourceConfig` I would suggest to rename it to `SideLoadedSubtitleTrack`.

Reply:
Yes, this is needed to guarantee that the end-user of the library won't be able to create a subtitle track without explicitly declaring the marked properties (url, label and language). And yes, I can include `mimeType` for Android, even though I don't know exactly which mime types are supported. I couldn't find anything related to that on the docs.

I tried to stick with the format types explicitly stated in [the iOS docs](https://cdn.bitmovin.com/player/ios/3/docs/Enums/SubtitleFormat.html#/c:@M@BitmovinPlayer@E@BMPSubtitleFormat) and then transform them back-and-fourth to their mime type representation:

- `vtt` -> `text/vtt`
- `ttml` -> `text/ttml`
- `cea` -> `text/cea`

This way we can keep a common interface between iOS and Android.

Anyways, I was hoping you could clarify to me exactly what formats/mime-types are supported in Android, and if they differ a lot, then it's the case of breaking up this common interface.

Reply:
@matamegger, there's one more thing.

I don't know if it's a difference in the SDK version, but I was able to apply and use subtitle tracks without passing a mime-type. Maybe you'd want to check this out.

Reply:
I think that's possible, I also have to admit that the API has it flaws in communicating its requirements and capabilities.
Probably the subtitle track was not shown in the list of `availableSubtitleTracks`, depending on your configuration, it might has shown, but it will not download anything. In that case it is just a UI gimick, that allows for external subtitle handling (e.g. for types we do not support).